### PR TITLE
1009 - Included the Feedback Webcomponent with the SSI

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves 1009

### 

# Explain the feature
Included the feedback component with the SSI of One Platform

## Does this PR introduce a breaking change

Yes

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

**Comment by riginoommen** [comment]
THis is a draft PR for the code review communication with @deshmukhmayur. I will update the template accordingly

File: packages/ssi-service/src/default.html [hunk 1]
```diff
@@ -68,12 +71,15 @@
...
       } );
     }
 
     const opfNav = document.createElement( "op-nav" );
-    const opfFeedback = document.createElement( "op-feedback" );
+    document.write('<opc-feedback></opc-feedback>');
+    document.querySelector('opc-feedback').addEventListener('opc-feedback:submit', (event) => {
+      event.detail.data.createdBy = window.OpAuthHelper.getUserInfo().rhatUUID;
+      window.sendFeedback( event.detail.data);
+    } );
```
Comment (deshmukhmayur) [inline packages/ssi-service/src/default.html:82]: Do not use document.write for adding html elements in the DOM. You can create the element using document.createElement and add it to the DOM using append or prepend.
```suggestion
    const opcFeedback = document.createElement('opc-feedback');
    opFeedback.on('opc-feedback:submit', 
    opFeedback.on('opc-feedback:submit', (event) => {
      event.detail.data.createdBy = window.OpAuthHelper?.getUserInfo()?.rhatUUID;
      window.sendFeedback( event.detail.data);
    } );
```

File: packages/ssi-service/src/default.html [hunk 2]
```diff
@@ -68,12 +71,15 @@
         }
         styleTag.remove();
         document.body.prepend( opfNav );
```
Comment (deshmukhmayur) [inline packages/ssi-service/src/default.html:73]:
```suggestion
        document.body.prepend( opfNav );
        document.body.append( opcFeedback );
```

File: packages/ssi-service/src/feedback-panel/api.js [hunk 3]
```diff
@@ -1,67 +1,49 @@
...
         }
-        return res.data;
-      } );
-  }
+    });
 }
 
-const APIServiceInstance = new APIService();
-export default APIServiceInstance;
+module.exports.sendFeedback = this.sendFeedback;
```
Comment (deshmukhmayur) [inline packages/ssi-service/src/feedback-panel/api.js:49]:
```suggestion
module.exports.sendFeedback = sendFeedback;

```

File: packages/ssi-service/src/feedback-panel/api.js [hunk 4]
```diff
@@ -1,67 +1,49 @@
...
+        variables: {
+            "input": feedbackInput
         }
-        return res.json();
-      } )
-      .then( res => {
-        if ( res.errors ) {
-          throw new Error( 'There were some errors in the query' + JSON.stringify( res.errors ) );
+    });
+    let xhr = new XMLHttpRequest();
```
Comment (deshmukhmayur) [inline packages/ssi-service/src/feedback-panel/api.js:30]: Can you replace this with a simple javascript fetch function?

Lookup [javascript fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch), xhr is very outdated and you don't have to construct an xhr object just to send a simple POST request.

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]

File: packages/ssi-service/src/feedback-panel/api.js [hunk 5]
```diff
@@ -1,67 +1,56 @@
...
         }
-        return res.data;
-      } );
-  }
+    });
+    return fetch(process.env.APPS_BASE_API, {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+            'Authorization': `${(window.OpAuthHelper.jwtToken) ? `Bearer ${window.OpAuthHelper.jwtToken}` : null}`
```
Comment (deshmukhmayur) [inline packages/ssi-service/src/feedback-panel/api.js:34]:
```suggestion
            'Authorization': `${(window.OpAuthHelper?.jwtToken) ? `Bearer ${window.OpAuthHelper.jwtToken}` : null}`
```

File: packages/ssi-service/src/feedback-panel/api.js [hunk 6]
```diff
@@ -1,67 +1,56 @@
...
+            } else if (!res.errors) {
+                (window.OpNotification) ? window.OpNotification.success({ subject: `Submitted Feedback`, link: response?.data?.createFeedback?.ticketUrl || null }) : alert('Submitted Feedback');
+            }
+            return res.data;
+        });
 }
 
-const APIServiceInstance = new APIService();
-export default APIServiceInstance;
+module.exports.sendFeedback = sendFeedback;
```
Comment (deshmukhmayur) [inline packages/ssi-service/src/feedback-panel/api.js:56]: Newline at the end of a file.
```suggestion
module.exports.sendFeedback = sendFeedback;

```

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]

**Review by deshmukhmayur — APPROVED** [review]
LGTM.

**Review by ghanlohar — APPROVED** [review]

**Review by ghanlohar — APPROVED** [review]
Resolve the conflicts before merging.

**Pull request merged**